fem/bilininteg.cpp

```diff
@@ -2665,13 +2665,13 @@ void VectorDiffusionIntegrator::AssembleElementVector(
       else if (MQ)
       {
          MQ->Eval(mcoeff, Tr, ip);
-         for (int i = 0; i < vdim; ++i)
+         for (int ii = 0; ii < vdim; ++ii)
```

Comment:
This seems counterproductive. Is `i` defined on the outside? Seems that the outside definition should change.

Reply:
Yes, the loop over integration points uses `i`. However, this is a very common choice (~86% of our bilinear form integrators use `i` for this loop iterator) so I'd rather not change the outer definition in this case.

Reply:
How about `vd` for the `vdim` iterator?

Reply:
We need two iterators because these are accessing entries of a matrix-valued coefficient. I suppose we could use `vi` and `vj` or `di` and `dj` but are these really any better? The entire nested loop is only a ten lines long and these iterators are each used in two places. I think there's little change of confusion.

Reply:
My personal preference would be `vd_i` and `vd_j` 🙂 

I agree there is not huge potential for confusion…

Reply:
If @vladotomov agrees I'll be happy to make that change.

Reply:
How about to make the loop over quadrature points `for (int q = 0 ...` and leave these `i` and `j`?
The outer index is used only at one place to do `ir->IntPoint(q)`

Reply:
I see your comment about the 86%, but don't you agree `q` is better in this case?

Reply:
I've used `q` in previous projects and it does seem natural to me. Unfortunately, `q` is more often used for coefficients in many MFEM classes.  The second most common iterator is `p` coming in at almost 11% (the only other one I see it `pint` which I don't like). It honestly doesn't matter much to me. My priorities were limiting the changes and/or following precedent. If @pazner is ok with using `q` (or `p`) either will be fine with me.

Reply:
Because this discussion thread did not have enough suggestions, here's another one.

How about getting rid of the outer index altogether (since it's only used once, to get the integration point), and using a range-based for loop:
```c++
for (const IntegrationPoint &ip : *ir)
{
   […]
}
```